com.unity.render-pipelines.high-definition/Editor/Lighting/LightmappingHDRP.cs

```diff
@@ -0,0 +1,156 @@
+        ///     * <see cref="Exception" /> if the <paramref name="probe" /> is not supported. Only
+        ///     <see cref="HDAdditionalReflectionData" /> probes are currently supported.
+        /// </returns>
+        public static Exception BakeProbe([NotNull] HDProbe probe, [NotNull] string path, BakeProbeOptions options)
+        {
+            // Check arguments
+            if (probe == null || probe.Equals(null)) return new ArgumentNullException(nameof(probe));
+            if (string.IsNullOrEmpty(path))
+                return new ArgumentException($"{nameof(path)} must not be empty or null.");
+
+            // We force RGBAHalf as we don't support 11-11-10 textures (only RT)
+            var probeFormat = GraphicsFormat.R16G16B16A16_SFloat;
+            switch (probe)
+            {
+                case HDAdditionalReflectionData _:
+                {
+                    // Get the texture size from the probe
+                    var textureSize = options.textureSize.Evaluate(probe);
+
+                    // Render and write
+                    var cubeRT = HDRenderUtilities.CreateReflectionProbeRenderTarget(textureSize, probeFormat);
+                    HDBakedReflectionSystem.RenderAndWriteToFile(probe, path, cubeRT, null);
+                    cubeRT.Release();
+
+                    // Import asset at target location
+                    AssetDatabase.ImportAsset(path);
+                    HDBakedReflectionSystem.ImportAssetAt(probe, path);
+
+                    // Assign to the probe the baked texture
+                    var bakedTexture = AssetDatabase.LoadAssetAtPath<Texture>(path);
+                    probe.SetTexture(ProbeSettings.Mode.Baked, bakedTexture);
+
+                    // Mark probe as dirty
+                    EditorUtility.SetDirty(probe);
+
+                    return null;
+                }
+                case PlanarReflectionProbe _:
+                    return new Exception("Planar reflection probe baking is not supported.");
+                default: return new Exception($"Cannot handle probe type: {probe.GetType()}");
+            }
+        }
+
+        /// <summary>
+        ///     Options of <see cref="LightmappingHDRP.BakeProbe" /> function.
+        /// </summary>
+        public struct BakeProbeOptions
+        {
+            /// <summary>
+            ///     Determines the size of the baked texture.
+            /// </summary>
+            public struct TextureSize
+            {
+                /// <summary>
+                ///     Which methods to use to evaluate the value.
+                /// </summary>
+                public enum Mode
+                {
+                    /// <summary>
+                    ///     Use the probe resolution as texture size.
+                    /// </summary>
+                    UseProbeResolution,
+
+                    /// <summary>
+                    ///     Use the value of <see cref="TextureSize.customValue" />.
```

Comment:
Uses the value of \<see cref="TextureSize.customValue" /> as the size of the baked texture.